lib/tracing-limit/src/lib.rs

```diff
@@ -10,199 +9,204 @@ use std::{
 
-            if let Some(limit) = limit_visitor.into_limit() {
-                let id = event.metadata().callsite();
+            Interest::sometimes()
+        } else {
+            Interest::always()
+        }
+    }
 
-                let events = self.events.read().unwrap();
-                if let Some((count, start)) = events.get(&id) {
-                    let local_count = count.fetch_add(1, Ordering::Relaxed);
-                    let start = start.load(Ordering::Relaxed);
-
-                    if ts - start < limit {
-                        // If this log is being limited and we've seen it for
-                        // a second time within that window. We will log out that
-                        // we are now rate limiting this log.
-                        //
-                        // If we have seen this event more than twice,
-                        // then lets early return to avoid passing this
-                        // event to the inner subscriber.
-                        if local_count == 1 {
-                            let mut visitor = FmtVisitor::default();
-                            event.record(&mut visitor);
-
-                            let message = if let Some(message) = &visitor.message {
-                                &message
-                            } else {
-                                "unknown event"
-                            };
-
-                            let meta = event.metadata();
-
-                            println!(
-                                "{} {} Limiting the output of \"{}\" for the next {} seconds",
-                                FmtLevel(&tracing_core::Level::INFO),
-                                meta.target(),
-                                message,
-                                limit
-                            );
-                            return;
-                        } else if local_count > 1 {
-                            return;
-                        }
-                    } else {
-                        drop(events);
-
-                        let mut events = self.events.write().unwrap();
-                        events.remove(&id);
-                        drop(events);
-
-                        let meta = event.metadata();
-
-                        let mut visitor = FmtVisitor::default();
-                        event.record(&mut visitor);
-
-                        let message = if let Some(message) = &visitor.message {
-                            &message
+    fn enabled(&self, metadata: &Metadata, ctx: Context<S>) -> bool {
+        if is_limited(metadata) {
+            let id = metadata.callsite();
+            let events = self.events.read().expect("lock poisoned!");
+
+            // check if the event exists within the map, if it does
+            // that means we are currently rate limiting it.
+            if let Some(state) = events.get(&id) {
+                let ts = SystemTime::now()
```

Comment:
I don't think an entire Unix timestamp is really necessary for this use case. Why not just use `Instant`s? You could make `state.limit` a `Duration` and `state.start` an `Instant`, and the test could just be:

```rust
if state.start.elapsed() < state.limit {
    // ...
}
```
Every call to `SystemTime::now` is _always_ a syscall, while `Instant::now` may not always require a syscall (depending on the platform); since we don't need the guarantees of a `SystemTime` here, it's probably better to use `Instant`s for performance.